libcloud/dns/drivers/powerdns.py

```diff
@@ -182,31 +189,47 @@ def create_record(self, name, zone, type, data, extra=None):
                                              zone.id)
         if extra is None or extra.get('ttl', None) is None:
             raise ValueError('PowerDNS requires a ttl value for every record')
-        record = {
+
+        record_name = self._fmt_name(name)
+        id = ':'.join((self.RECORD_TYPE_MAP[type], record_name))
```

Comment:
You are right, PowerDNS does not expose ID via HTTP api, however even Route53 does not have an ID, but the driver generates it with the same method. I've used the same technique because I found it useful.